src/test/regress/expected/foreign_key_to_reference_table_0.out

```diff
@@ -0,0 +1,1905 @@
+ referencing_referencing_table_id_fkey_7000361 | fkey_reference_table.referencing_referencing_table_7000361 | fkey_reference_table.referencing_table_7000353
+ referencing_referencing_table_id_fkey_7000362 | fkey_reference_table.referencing_referencing_table_7000362 | fkey_reference_table.referencing_table_7000354
+ referencing_referencing_table_id_fkey_7000363 | fkey_reference_table.referencing_referencing_table_7000363 | fkey_reference_table.referencing_table_7000355
+(16 rows)
+
+INSERT INTO referenced_table SELECT x, x+1 FROM generate_series(1,1000) AS f(x);
+INSERT INTO referencing_table SELECT x, x+1, x+2 FROM generate_series(1,999) AS f(x);
+INSERT INTO referencing_referencing_table SELECT x, x+1 FROM generate_series(1,999) AS f(x);
+DELETE FROM referenced_table WHERE test_column > 800;
+SELECT max(ref_id) FROM referencing_referencing_table;
+ max
+---------------------------------------------------------------------
+ 800
+(1 row)
+
+DROP TABLE referenced_table CASCADE;
+NOTICE:  drop cascades to constraint fkey_ref on table referencing_table
+DROP TABLE referencing_table CASCADE;
+NOTICE:  drop cascades to constraint referencing_referencing_table_id_fkey on table referencing_referencing_table
+DROP TABLE referencing_referencing_table;
+-- test if create_distributed_table works in transactions with some edge cases
+-- the following checks if create_distributed_table works on foreign keys when
+-- one of them is a self-referencing table of multiple distributed tables
+BEGIN;
+  CREATE TABLE test_table_1(id int PRIMARY KEY);
+  SELECT create_reference_table('test_table_1');
+ create_reference_table
+---------------------------------------------------------------------
+
+(1 row)
+
+  CREATE TABLE test_table_2(id int PRIMARY KEY, value_1 int, FOREIGN KEY(id) REFERENCES test_table_1(id));
+  SELECT create_distributed_table('test_table_2', 'id');
+NOTICE:  creating a new table for fkey_reference_table.test_table_2
+NOTICE:  Moving the data of fkey_reference_table.test_table_2
+NOTICE:  Dropping the old fkey_reference_table.test_table_2
+NOTICE:  Renaming the new table to fkey_reference_table.test_table_2
+ create_distributed_table
+---------------------------------------------------------------------
+
+(1 row)
+
+  CREATE TABLE test_table_3(id int PRIMARY KEY, value_1 int, FOREIGN KEY(value_1) REFERENCES test_table_1(id), FOREIGN KEY(id) REFERENCES test_table_2(id));
+ERROR:  cannot create foreign key: cannot create foreign key constraint since foreign keys from reference tables and citus local tables to distributed tables are not supported
+DETAIL:  Reference tables and citus local tables can only have foreign keys to reference tables and citus local tables
+  SELECT create_distributed_table('test_table_3', 'id');
+ERROR:  current transaction is aborted, commands ignored until end of transaction block
+  DROP TABLE test_table_1 CASCADE;
+ERROR:  current transaction is aborted, commands ignored until end of transaction block
+ROLLBACK;
```

Comment:
Now added an alternative test output for foreign_key_to_reference_table.
This is because we run this test in multi_schedule two times: addding and without adding coordiantor to metadata.
This `_0` test output is for the case where we have coordinator in metadata.

-----

I will look into marked lines to understand why we give this error when coordinator is added to coordinator

Reply:
Understood the reason why:

Now we create `test_table_1` that references to `test_table_2` (reference table) and `test_table_3` (distributed table).
As we are defining foreign key to reference table `test_table_2`, we convert `test_table_1` to a citus local table.

However, when creating citus local table from `test_table_1`, we give
>ERROR:  cannot create foreign key: cannot create foreign key constraint since foreign keys from reference tables and citus local tables to distributed tables are not supported
DETAIL:  Reference tables and citus local tables can only have foreign keys to reference tables and citus local tables

error when re-defining foreign key to distributed table.

So this is the expected output.

Reply:
apart from that, also see https://github.com/citusdata/citus/pull/4479#discussion_r556931518